google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/HelloWorld.java

```diff
@@ -0,0 +1,204 @@
+  private static final String columnFamily = "cf1";
+  private static final String columnQualifier = "greeting";
+  private static final String rowKeyPrefix = "rowKey";
+
+  public static void main(String[] args) throws Exception {
+
+    if (args.length != 2) {
+      System.out.println("Missing required project id or instance id");
+    }
+    projectId = args[0];
+    instanceId = args[1];
+
+    HelloWorld helloWorld = new HelloWorld(projectId, instanceId, generateTableId());
+    helloWorld.run();
+  }
+
+  public HelloWorld(String projectId, String instanceId, String tableId) throws IOException {
+    this.projectId = projectId;
+    this.instanceId = instanceId;
+    this.tableId = tableId;
+
+    // [START connecting_to_bigtable]
+    // Create the settings to configure a bigtable data client
+    BigtableDataSettings settings =
+        BigtableDataSettings.newBuilder()
+            .setInstanceName(InstanceName.of(projectId, instanceId))
+            .build();
+
+    // Create bigtable data client
+    dataClient = BigtableDataClient.create(settings);
+
+    // Create the settings to configure a bigtable admin client
+    BigtableTableAdminSettings adminSettings =
+        BigtableTableAdminSettings.newBuilder()
+            .setInstanceName(com.google.bigtable.admin.v2.InstanceName.of(projectId, instanceId))
+            .build();
+
+    // Create bigtable admin client
+    adminClient = BigtableTableAdminClient.create(adminSettings);
+    // [END connecting_to_bigtable]
+  }
+
+  public void run() throws Exception {
+    try {
+      createTable();
+      writeToTable();
+      readSingleRow();
+      readTable();
+      deleteTable();
+      garbageCollect();
+    } catch (ApiException e) {
+      System.err.println("Exception while running HelloWorld: " + e.getMessage());
+    } finally {
+      dataClient.close();
+      adminClient.close();
+    }
+  }
+
+  public void createTable() {
+    // [START creating_a_table]
+    // Check if table exists, create table if does not exist
+    if (!adminClient.exists(tableId)) {
+      CreateTableRequest createTableRequest =
+          CreateTableRequest.of(tableId).addFamily(columnFamily);
+      System.out.println("Creating table: " + tableId);
+      adminClient.createTable(createTableRequest);
+      System.out.printf("Table %s created successfully%n", tableId);
+    }
+    // [END creating_a_table]
+  }
+
+  public void writeToTable() {
+    // [START writing_rows]
+    try {
+      System.out.println("\nWriting some greetings to the table");
+      String[] greetings = {"Hello World!", "Hello Bigtable!", "Hello Java!"};
+      for (int i = 0; i < greetings.length; i++) {
+        RowMutation rowMutation =
+            RowMutation.create(tableId, rowKeyPrefix + i)
+                .setCell(columnFamily, columnQualifier, greetings[i]);
+        dataClient.mutateRow(rowMutation);
+        System.out.println(greetings[i]);
+      }
+    } catch (ApiException e) {
```

Comment:
It's not obvious that the client throws typed exceptions for different status codes. So I thought it would be good to show a sample. The javadocs for the data client show a couple of examples:

https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java#L450

https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java#L543

Other subclasses are: NotFoundException, DeadlineExceededException, PermissionDeniedException, AlreadyExistsException